test/e2e/utils.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"time"
+
+	"github.com/golang/glog"
+	"golang.org/x/oauth2/google"
+	cloudresourcemanager "google.golang.org/api/cloudresourcemanager/v1"
+	boskosclient "k8s.io/test-infra/boskos/client"
+	remote "sigs.k8s.io/gcp-compute-persistent-disk-csi-driver/test/binremote"
+)
+
+var (
+	boskos = boskosclient.NewClient(os.Getenv("JOB_NAME"), "http://boskos")
+)
+
+const (
+	archiveName = "e2e_gce_pd_test.tar.gz"
+)
+
+func setupInstanceAndDriver(instanceProject, instanceZone, instanceName, port, instanceServiceAccount string) (*remote.InstanceInfo, error) {
+	// Create the instance in the requisite zone
+	instance, err := remote.CreateInstanceInfo(instanceProject, instanceZone, instanceName)
+	if err != nil {
+		return nil, err
+	}
+
+	err = instance.CreateInstance(instanceServiceAccount)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Create Driver Archive
+
+	goPath, ok := os.LookupEnv("GOPATH")
+	if !ok {
+		return nil, fmt.Errorf("Could not find environment variable GOPATH")
+	}
+	pkgPath := path.Join(goPath, "src/sigs.k8s.io/gcp-compute-persistent-disk-csi-driver/")
+	binPath := path.Join(pkgPath, "bin/gce-pd-csi-driver")
+	archivePath, err := remote.CreateDriverArchive(archiveName, pkgPath, binPath)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err = os.Remove(archivePath)
+		if err != nil {
+			glog.Warningf("Failed to remove archive file %s: %v", archivePath, err)
+		}
+	}()
+
+	// Upload archive to instance and run binaries
+	endpoint := fmt.Sprintf("tcp://localhost:%s", port)
+	workspace := remote.NewWorkspaceDir("gce-pd-e2e-")
+	driverRunCmd := fmt.Sprintf("sh -c '/usr/bin/nohup %s/gce-pd-csi-driver --endpoint=%s --nodeid=%s > %s/prog.out 2> %s/prog.err < /dev/null &'",
+		workspace, endpoint, instanceName, workspace, workspace)
+	err = instance.UploadAndRun(archivePath, workspace, driverRunCmd)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create an SSH tunnel from port to port
+	res, err := instance.CreateSSHTunnel(port, port)
+	if err != nil {
+		return nil, fmt.Errorf("SSH Tunnel pid %v encountered error: %v", res, err)
+	}
+
+	return instance, nil
+}
+
+func setupProwConfig() (project, serviceAccount string) {
```

Comment:
Are all of these util methods being used?

Reply:
can you separate out utils into a separate e2e/util pkg? Then that would make it easy for me to import